proto/src/lib.rs

```diff
@@ -1,60 +1,28 @@
-use std::{env, fs, path::PathBuf};
+use std::{env, path::PathBuf};
 
-use anyhow::Context;
-use prost::Message;
+use tonic_build::FileDescriptorSet;
 
 const RPC_PROTO: &str = "rpc.proto";
-const STORE_PROTO: &str = "store.proto";
-const BLOCK_PRODUCER_PROTO: &str = "block_producer.proto";
-
-pub struct ProtoBuilder(tonic_build::Builder);
 
-impl ProtoBuilder {
-    /// Creates a new `Builder` from the provided `tonic_build::Builder`.
-    /// The settings on the `Builder` are used to compile the proto files, including the `out_dir`
-    /// directory.
-    pub fn new(builder: tonic_build::Builder) -> Self {
-        Self(builder)
-    }
+#[cfg(feature = "internal")]
+const STORE_PROTO: &str = "store.proto";
 
-    /// Compiles the server proto bindings. Generated files are written to the `out_dir` set on the
-    /// internal Builder.
-    ///
-    /// Generates bindings for the store, block-producer, and rpc server components.
-    pub fn compile_server(self) -> anyhow::Result<()> {
-        generate_protos(self.0.clone(), &[STORE_PROTO, BLOCK_PRODUCER_PROTO])?;
-        generate_protos(self.0.build_client(false), &[RPC_PROTO])
-    }
+#[cfg(feature = "internal")]
+const BLOCK_PRODUCER_PROTO: &str = "block_producer.proto";
 
-    /// Compiles the client proto bindings. Generated files are written to the `out_dir` set on the
-    /// internal Builder.
-    ///
-    /// Generates bindings for the rpc client component.
-    pub fn compile_rpc_client(self) -> anyhow::Result<()> {
-        generate_protos(self.0.build_client(true), &[RPC_PROTO])
-    }
+pub fn rpc_file_descriptor() -> Result<FileDescriptorSet, protox::Error> {
+    let proto_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("proto");
+    protox::compile([RPC_PROTO], &[proto_dir])
 }
 
-/// This wrapper function receives a `tonic_build::Builder` and uses it to generate the protobuf
-/// bindings.
-fn generate_protos(builder: tonic_build::Builder, proto_files: &[&str]) -> anyhow::Result<()> {
-    let out = env::var("OUT_DIR").context("env::OUT_DIR not set")?;
-    let file_descriptor_path = PathBuf::from(out).join("file_descriptor_set.bin");
-
-    let crate_root: PathBuf = env!("CARGO_MANIFEST_DIR").into();
-    let proto_dir = crate_root.join("proto");
-
-    let includes = &[proto_dir];
-    let file_descriptors = protox::compile(proto_files, includes)?;
-    fs::write(&file_descriptor_path, file_descriptors.encode_to_vec())
-        .context("writing file descriptors")?;
-
-    let mut prost_config = prost_build::Config::new();
-    prost_config.skip_debug(["AccountId", "Digest"]);
+#[cfg(feature = "internal")]
+pub fn store_file_descriptor() -> Result<FileDescriptorSet, protox::Error> {
```

Comment:
I think these shouldn't return a `Result` - we can just `.expect` since we know the files are good.